tests/Doctrine/Tests/ORM/Functional/Ticket/GH7804.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Annotation as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class GH7804 extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH7804Entity::class,
+        ]);
+    }
+
+    public function testPositiveQuery(): void
+    {
+        $qb = $this->em->createQueryBuilder();
+
+        $qb->from(GH7804Entity::class, 'te1')
+            ->select('te1.title as t')
+            ->addGroupBy('t');
+
+        self::assertSQLEquals(
```

Comment:
Please test the result of the query instead of the generated query to avoid platform incompatibility issues.